Modules/_testcapimodule.c

```diff
@@ -3444,6 +3453,7 @@ static PyMethodDef TestMethods[] = {
     {"function_set_closure", function_set_closure, METH_VARARGS, NULL},
     {"check_pyimport_addmodule", check_pyimport_addmodule, METH_VARARGS},
     {"test_weakref_capi", test_weakref_capi, METH_NOARGS},
+    {"test_warn", test_warn, METH_NOARGS},
```

Comment:
AFAIK, all functions in the `_testcapi` module that starts with `test_` are ran as separate tests, so `test_warn` will always run and emit a warning in tests. At least it used to be.

So it is better to use other name. In future we may want to add more complex wrapper for `PyErr_WarnEx()` which accepts arguments.

Reply:
It is still true. See `get_test_funcs()` in `Lib/test/test_capi/test_misc.py`.

Reply:
Not right now. But in future this function will be replaced.